src/object.cr

```diff
@@ -495,7 +495,11 @@ class Object
         end
 
         def {{method_prefix}}\{{name.id}}
```

Comment:
How about adding the not nil return type?  Is helpful for documentation.

Reply:
It's not always available and IMO seems like a bit redundant piece of information.

Reply:
If its provided it should be included TBH.

Reply:
I can add it, but would like a definite answer on whether it should be added.

Reply:
It also looks like there might be a bug with the existing type annotations for these macros. Using a union of types puts the nillable `?` on the last type instead of the whole thing.

```crystal
getter? name : String | Symbol

# produces:

@name : String | Symbol?

def name? # No type annotation
  @name
end

# should be:

@name : (String | Symbol)?

def name? : (String | Symbol)?
  @name
end
```

Reply:
Note that `X | Y?` is exactly the same as `X | Y | Nil`

Reply:
Yea the `?` getter should include `Nil` in the return type.  The non `?` getter should exclude the `Nil` type in the restriction.

If it's provided, i.e. the `TypeDeclaration` context, I see no reason not to include them.  The method is already inherently that type, i.e. you can't add custom logic into it.  It will just help API docs since it would now include the expected types.

Reply:
@bew Oh, duh, you're right. It doesn't matter if it's in parens or not.

@Blacksmoke16 I agree. If the type annotation is there, it should be applied to the getter methods. I'll add that in for all related macros.